core/src/main/scala/akka/persistence/jdbc/state/javadsl/DurableStateStore.scala

```diff
@@ -0,0 +1,55 @@
+package akka.persistence.jdbc.state.javadsl
+
+import java.util.Optional
+import java.util.concurrent.CompletionStage
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.{ ExecutionContext, Future }
+import slick.jdbc.{ JdbcBackend, JdbcProfile }
+import akka.Done
+import akka.persistence.state.javadsl.{ DurableStateUpdateStore, GetObjectResult }
+import akka.serialization.Serialization
+import akka.persistence.jdbc.state.{ AkkaSerialization, DurableStateQueries }
+import akka.persistence.jdbc.config.DurableStateTableConfiguration
+import akka.persistence.jdbc.state.DurableStateTables
+import akka.dispatch.ExecutionContexts
+
+class DurableStateStore[A](
+    db: JdbcBackend#Database,
+    profile: JdbcProfile,
+    durableStateConfigConfig: DurableStateTableConfiguration,
+    serialization: Serialization)(implicit ec: ExecutionContext)
+    extends DurableStateUpdateStore[A] {
+  import profile.api._
+
+  val queries = new DurableStateQueries(profile, durableStateConfigConfig)
+
+  def getObject(persistenceId: String): CompletionStage[GetObjectResult[A]] =
+    toJava(db.run(queries._selectByPersistenceId(persistenceId).result).map { rows =>
+      rows.headOption match {
+        case Some(row) =>
+          GetObjectResult(
+            Optional
+              .ofNullable(AkkaSerialization.fromRow(serialization)(row).toOption.getOrElse(null))
+              .asInstanceOf[Optional[A]],
+            row.seqNumber)
+
+        case None =>
+          throw new Exception(s"State object creation failed during fetch from store: persistenceId $persistenceId")
+      }
+    })
```

Comment:
Instead of implementing those twice for javadsl and scaladsl it should delegate to the scaladsl implementation and only do the conversions here. `scaladsl.DurableStateStore` can be passed as constructor parameter and those methods can delegate to that.

Reply:
ok, didn't want to make `javadsl` depend on `scaladsl` package. But will change ..

Reply:
done ..